Dockerfile

```diff
@@ -0,0 +1,39 @@
+# Use the Python 3.11 slim image as the base image
+FROM python:3.11-slim
+
+# Set the working directory inside the container
+WORKDIR /app
+
+# Enable non-interactive mode for debconf
+RUN echo 'debconf debconf/frontend select Noninteractive' | debconf-set-selections
+
+# Update the package repository and install required dependencies
+RUN apt-get update -y && \
+    apt-get install -y build-essential cmake unzip pkg-config \
+    libjpeg-dev libpng-dev libtiff-dev \
+    libavcodec-dev libavformat-dev libswscale-dev libv4l-dev \
+    libatlas-base-dev gfortran && \
+    apt-get clean && \
+    apt-get -y autoremove
+
+# Install Python dependencies
+RUN pip3 install --user opencv-python opencv-contrib-python
+
+# Copy over the development and production requirements files
+COPY ./requirements.dev.txt /app
+COPY ./requirements.txt /app
+
+# Install additional Python dependencies for development
+RUN pip3 install -r requirements.dev.txt
+
+# Copy the entire project into the container
+COPY . /app
```

Comment:
How about excluding samples and tests folder in a Dockerignore file?

Reply:
Alright, time to play hide and seek with Docker! Let's give those pesky 'samples' and 'tests' folders a timeout in the Dockerignore file. No peeking allowed!